src/lib.rs

```diff
@@ -392,16 +383,9 @@ impl<'a> ReportErrorBuilder<'a> {
     add_generic_field!(with_title, title, Into<String>);
 
     // Send the report to Rollbar.
-    pub fn send(&mut self) -> thread::JoinHandle<Option<ResponseStatus>> {
+    pub fn send(&mut self) -> Option<ResponseStatus> {
         let client = self.report_builder.client;
-
-        match self.report_builder.send_strategy {
-            Some(ref send_strategy) => {
-                let http_client = client.http_client.to_owned();
-                send_strategy(http_client, self.to_string())
-            }
-            None => client.send(self.to_string()),
-        }
+        block_on(client.send(self.to_string()))
```

Comment:
once you introduce async await, the whole call chain up to main() needs to be async, but i found that you could use block_on to get out of async world. the caller of this function is immediately going to call join() on the join handle anyway, so it should change anything to block on the async here, right?